components/director/internal/repo/delete.go

```diff
@@ -30,12 +38,14 @@ func (g *Deleter) Delete(ctx context.Context, tenant string, conditions Conditio
 	if err != nil {
 		return errors.Wrap(err, "while deleting from database")
 	}
-	affected, err := res.RowsAffected()
-	if err != nil {
-		return errors.Wrap(err, "while checking affected rows")
-	}
-	if affected != 1 {
-		return fmt.Errorf("delete should remove single row, but removed %d rows", affected)
+	if requireSingleRemoval {
```

Comment:
Invert the condition
```
if !requireSingleRemoval {
   return nil
}
```

Reply:
discussed already, the current version seems to be more readable.